src/nbl/video/COpenGLCommandBuffer.cpp

```diff
@@ -0,0 +1,743 @@
+                // TODO
+            }
+            break;
+            case ECT_BEGIN_RENDERPASS:
+            {
+                auto& c = cmd.get<ECT_BEGIN_RENDERPASS>();
+                auto framebuf = core::smart_refctd_ptr_static_cast<const COpenGLFramebuffer>(c.renderpassBegin.framebuffer);
+
+                ctxlocal->nextState.framebuffer.hash = framebuf->getHashValue();
+                ctxlocal->nextState.framebuffer.fbo = std::move(framebuf);
+                ctxlocal->flushStateGraphics(gl, SOpenGLContextLocalCache::GSB_FRAMEBUFFER, ctxid);
+
+                GLuint fbo = ctxlocal->currentState.framebuffer.GLname;
+                beginRenderpass_clearAttachments(gl, c.renderpassBegin, fbo);
+            }
+            break;
+            case ECT_NEXT_SUBPASS:
+            {
+                auto& c = cmd.get<ECT_NEXT_SUBPASS>();
+                // TODO (when we support subpasses) some barriers based on subpass dependencies?
+                // not needed now tho, we dont support multiple subpasses yet
+            }
+            break;
+            case ECT_END_RENDERPASS:
+            {
+                auto& c = cmd.get<ECT_END_RENDERPASS>();
+                // no-op
+            }
+            break;
+            case ECT_SET_DEVICE_MASK:
+            {
+                auto& c = cmd.get<ECT_SET_DEVICE_MASK>();
+                // no-op
+            }
+            break;
+            case ECT_BIND_GRAPHICS_PIPELINE:
+            {
+                auto& c = cmd.get<ECT_BIND_GRAPHICS_PIPELINE>();
+
+                ctxlocal->updateNextState_pipelineAndRaster(c.pipeline->getRenderpassIndependentPipeline(), ctxid);
+            }
+            break;
+            case ECT_BIND_COMPUTE_PIPELINE:
+            {
+                auto& c = cmd.get<ECT_BIND_COMPUTE_PIPELINE>();
+
+                const COpenGLComputePipeline* glppln = static_cast<const COpenGLComputePipeline*>(c.pipeline.get());
+                ctxlocal->nextState.pipeline.compute.usedShader = glppln ? glppln->getShaderGLnameForCtx(0u, ctxid) : 0u;
+                ctxlocal->nextState.pipeline.compute.pipeline = core::smart_refctd_ptr<const COpenGLComputePipeline>(glppln);
+            }
+            break;
+            case ECT_RESET_QUERY_POOL:
+            {
+                auto& c = cmd.get<ECT_RESET_QUERY_POOL>();
+            }
+            break;
+            case ECT_BEGIN_QUERY:
+            {
+                auto& c = cmd.get<ECT_BEGIN_QUERY>();
+            }
+            break;
+            case ECT_END_QUERY:
+            {
+                auto& c = cmd.get<ECT_END_QUERY>();
+            }
+            break;
+            case ECT_COPY_QUERY_POOL_RESULTS:
+            {
+                auto& c = cmd.get<ECT_COPY_QUERY_POOL_RESULTS>();
+            }
+            break;
+            case ECT_WRITE_TIMESTAMP:
+            {
+                auto& c = cmd.get<ECT_WRITE_TIMESTAMP>();
+            }
+            break;
```

Comment:
you should really make a `NBL_TODO` macro that just expands to `NBL_DEBUG_BREAK_IF`